pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SnapshotSegmentAbortedTxnProcessorImpl.java

```diff
@@ -0,0 +1,600 @@
+                    .thenRun(() -> {
+                        persistentSnapshotIndexes = snapshotIndexes;
+                        lastSnapshotTimestamps = System.currentTimeMillis();
+                    })
+                    .exceptionally(e -> {
+                        log.error("[{}] Failed to update snapshot segment index", snapshotIndexes.getTopicName(), e);
+                        return null;
+                    });
+        }
+
+        //Only update the metadata in the transactionBufferSnapshotIndexes.
+        private CompletableFuture<Void> updateIndexMetadataForTheLastSnapshot(PositionImpl maxReadPosition,
+                                                                              ArrayList<TxnID> abortedTxns) {
+            TransactionBufferSnapshotIndexesMetadata metadata = new TransactionBufferSnapshotIndexesMetadata(
+                    maxReadPosition.getLedgerId(), maxReadPosition.getEntryId(), serializationForSegment(abortedTxns));
+
+            return updateSnapshotIndex(metadata, persistentSnapshotIndexes.getIndexList());
+        }
+
+        protected CompletableFuture<Void> clearSnapshotSegmentAndIndexes() {
+            ArrayList<CompletableFuture<Void>> completableFutures = new ArrayList<>();
+            //Delete all segment
+            while (!abortTxnSegments.isEmpty()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("[{}] Topic transaction buffer clear aborted transactions, maxReadPosition : {}",
+                            topic.getName(), abortTxnSegments.firstKey());
+                }
+                PositionImpl positionNeedToDelete = abortTxnSegments.firstKey();
+                completableFutures.add(persistentWorker.deleteSnapshotSegment(positionNeedToDelete));
+            }
+            //Delete index
+            return FutureUtil.waitForAll(completableFutures)
+                    .thenCompose((ignore) -> snapshotIndexWriterFuture
+                            .thenCompose(indexesWriter -> indexesWriter.writeAsync(topic.getName(), null)))
+                    .thenRun(() -> {
+                        log.info("Successes to clear the snapshot segment and indexes for the topic [{}]",
+                                topic.getName());
+
+                    })
+                    .exceptionally(e -> {
+                        log.error("Failed to clear the snapshot segment and indexes for the topic [{}]",
+                                topic.getName(), e);
+
+                        return null;
+                    });
+        }
+
+
+        CompletableFuture<Void> closeAsync() {
+            return CompletableFuture.allOf(
+                    this.snapshotIndexWriterFuture.thenCompose(SystemTopicClient.Writer::closeAsync),
+                    this.snapshotSegmentsWriterFuture.thenCompose(SystemTopicClient.Writer::closeAsync));
+        }
+    }
+
+    ArrayList<TxnID> deserializationFotSnapshotSegment(List<TxnIDData> snapshotSegment) {
```

Comment:
Why not this method with the modifier `private`?

Same for `serializationForSegment `